### PR TITLE
Custom FileSystem Support

### LINKED ISSUE #9: Custom FileSystem fails
Could you change the following line of the CreateScriptServices method:
``` cs
scriptServicesBuilder.Overrides[typeof(IFileSystem)] = _options.FileSystem;
```
to
``` cs
initializationServices.Overrides[typeof(IFileSystem)] = _options.FileSystem;
```
because the `LoadModules` method of the `ScriptServicesBuilder` calls `InitializationServices.GetFileSystem()`
Thanks

### PR DESCRIPTION
**Pull request by beolutz**
Fix #9 - Custom FileSystem Support
(Works as previously discussed)

### CODE REVIEW

**Comment by laedit** [comment]
Thanks for the PR!
I will look at why AppVeyor didn't launch build.

**Comment by laedit** [comment]
It seems that the only way is to close and re-open the PR.

**Comment by beolutz** [comment]
You’re welcome.

**Comment by laedit** [comment]
Well, I need to fix an issue in the build process since AppVeyor consider that [tests failing](https://ci.appveyor.com/project/laedit/scriptcs-mef/build/0.2.2-PullRequest.11+2.build.60/tests) is ok.
So this PR cannot be merged until all tests are green and like I said in [#9](https://github.com/scriptcs-contrib/scriptcs-mef/issues/9#issuecomment-265662388) it seems that there a bug in the support of custom filesystem in hosting scenario.
@glennblock can you help us on this?
Thanks a lot.

File: src/ScriptCs.ComponentModel.Composition.Test/ScriptCsCatalogFacts.cs [hunk 1]
```diff
@@ -755,6 +754,8 @@ private static Mock<IFileSystem> GetMockFileSystem(string[] fileNames, string[]
             fileSystem.SetupGet(f => f.BinFolder).Returns("scriptcs_bin");
             fileSystem.SetupGet(f => f.DllCacheFolder).Returns(".scriptcs_cache");
             fileSystem.SetupGet(f => f.NugetFile).Returns("scriptcs_nuget.config");
+            fileSystem.SetupGet(f => f.GlobalFolder).Returns(Environment.CurrentDirectory);
```
Comment (laedit) [inline src/ScriptCs.ComponentModel.Composition.Test/ScriptCsCatalogFacts.cs]: I am not comfortable with using `Environment` in a unit test, is this mandatory?

**Review by laedit — COMMENTED** [review]
I haven't succeeded in using a full custom `IFilSystem`, did you?
If yes can you share it?
I am trying to determine if the issue is from ScriptCS or the hosting part here.

**Comment by laedit** [comment]
Sorry for my last comment, my implementation of `IFileSystem`was faulty.
So the `HostBin` must be the real path of the host and cannot be _virtualized_, but can you replace the `Environement.CurrentDirectory` of `GlobalFolder` in the test by `@"c:\workingdirectory"`?
Thanks.

**Comment by beolutz** [comment]
Should be good now.
Any Chance to get a nuget package update anytime soon?

**Comment by laedit** [comment]
Thanks.
I will release a new version ASAP, at the latest this sunday.

**Review by laedit — APPROVED** [review]

**Pull request merged**